.github/workflows/integration_tests.yml

```diff
@@ -384,18 +347,403 @@ jobs:
+        uses: phulsechinmay/rewritable-pr-comment@v0.3.0
+        if: ${{ needs.check_trigger.outputs.should_update_labels && failure() && !cancelled() }}
+        with:
+          message: |
+            ### ❌&nbsp; Integration test FAILED (but still ⏳&nbsp; in progress)
+            Requested by @${{github.actor}} on commit ${{needs.prepare_matrix.outputs.github_ref}}
+            Last updated: ${{ steps.get-time.outputs.time }}
+            **[View integration test results](https://github.com/${{github.repository}}/actions/runs/${{github.run_id}})**
+            ${{ env.SUMMARY_TABLE }}
+          GITHUB_TOKEN: ${{ github.token }}
+          COMMENT_IDENTIFIER: ${{ env.statusCommentIdentifier }}
+          ISSUE_ID: ${{needs.check_trigger.outputs.pr_number}}
+
+      - name: Summarize build results
+        if: ${{ !cancelled() }}
+        shell: bash
+        run: |
+          cat build-results-ios-macos-latest.log
+          if [[ "${{ job.status }}" != "success" ]]; then
+            exit 1
+          fi
+
+  test_desktop:
+    name: test-desktop-${{ matrix.os }}-${{ matrix.ssl_variant }}
+    needs: [check_trigger, prepare_matrix, build_desktop]
+    runs-on: ${{ matrix.os }}
+    if: contains(needs.prepare_matrix.outputs.matrix_platform, 'Desktop') && needs.prepare_matrix.outputs.apis != '' && !cancelled() && !failure()
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ${{ fromJson(needs.prepare_matrix.outputs.matrix_os) }}
+        ssl_variant: ${{ fromJson(needs.prepare_matrix.outputs.matrix_ssl) }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{needs.prepare_matrix.outputs.github_ref}}
+      - name: download artifact
+        uses: actions/download-artifact@v2.0.8
+        with:
+          path: testapps
+          name: testapps-desktop-${{ matrix.os }}-${{ matrix.ssl_variant }}.zip
+      - name: unzip artifact
+        run: |
+          unzip testapps/testapps-desktop-${{ matrix.os }}-${{ matrix.ssl_variant }}.zip
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+      - name: Install python deps
+        run: |
+          pip install -r scripts/gha/requirements.txt
+          python scripts/gha/restore_secrets.py --passphrase "${{ secrets.TEST_SECRET }}" --artifact testapps-desktop-${{ matrix.os }}-${{ matrix.ssl_variant }}
+      - name: Run desktop integration tests
+        run: |
+          python scripts/gha/desktop_tester.py --testapp_dir "testapps-desktop-${{ matrix.os }}-${{ matrix.ssl_variant }}" --artifact_name "desktop-${{ matrix.os }}-${{ matrix.ssl_variant }}"
+      - name: Upload Desktop test results artifact
+        if: ${{ !cancelled() }}
+        uses: actions/upload-artifact@v2.2.2
+        with:
+          name: log-artifact
+          path: testapps-desktop-${{ matrix.os }}-${{ matrix.ssl_variant }}/test-results-desktop-${{ matrix.os }}-${{ matrix.ssl_variant }}.log
```

Comment:
I think "name" here is supposed to be the name of the artifact, not the name of the workflow step, right?

Reply:
The "name" here is not the name of artifact, it's more like a dir. The name of artifact is extracted from the "path". I was also confused at the beginning.